model-optimizer/mo/main.py

```diff
@@ -97,16 +97,25 @@ def print_argv(argv: argparse.Namespace, is_caffe: bool, is_tf: bool, is_mxnet:
 
 
 def prepare_ir(argv: argparse.Namespace):
-    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx = deduce_framework_by_namespace(argv)
-
     fem = argv.feManager
-    new_front_ends = []
-    if fem is not None:  # in future, check of 'use_legacy_frontend' in argv can be added here
-        new_front_ends = fem.get_available_front_ends()
+    available_moc_front_ends = []
+    moc_front_end = None
+    if fem:  # in future, check of 'use_legacy_frontend' in argv can be added here
+        available_moc_front_ends = fem.get_available_front_ends()
+        if argv.input_model:
+            if not argv.framework:
+                moc_front_end = fem.load_by_model(argv.input_model)
+                if moc_front_end:
+                    argv.framework = moc_front_end.get_name()
+            elif argv.framework in available_moc_front_ends:
+                moc_front_end = fem.load_by_framework(argv.framework)
+
+    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx =\
+        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False]
```

Comment:
not clear why you set all to `False`. I think you need to do all this manipulations with framework detection in `deduce_framework_by_namespace` routine. It is more logical.

Reply:
In this case I'll need to return 'available_front_ends' and loaded 'moc_front_end' from this function to reuse it further.
I thought it will complicate `deduce_framework_by_namespace` and also it's name is not intended to load anything. So, this function is left as is (as legacy to recognize only python frontends) and is called only when suitable MOC frontend is not found.